src/io.c

```diff
@@ -406,15 +406,34 @@ dispatch_io_create_f(dispatch_io_type_t type, dispatch_fd_t fd,
 			^(int error){ cleanup_handler(context, error); });
 }
 
+#if defined(_WIN32)
+#define is_separator(ch) ((ch) == '/' || (ch) == '\\')
+#else
+#define is_separator(ch) ((ch) == '/')
+#endif
+
 dispatch_io_t
 dispatch_io_create_with_path(dispatch_io_type_t type, const char *path,
 		int oflag, mode_t mode, dispatch_queue_t queue,
 		void (^cleanup_handler)(int error))
 {
-	if ((type != DISPATCH_IO_STREAM && type != DISPATCH_IO_RANDOM) ||
-			!(*path == '/')) {
+	if (type != DISPATCH_IO_STREAM && type != DISPATCH_IO_RANDOM) {
+		return DISPATCH_BAD_INPUT;
+	}
+#if defined(_WIN32)
+	// Absolute paths begin with either a separator or a character followed by a
+	// colon and a separator
+	if (path[0] == '\0') {
+		return DISPATCH_BAD_INPUT;
+	}
+	if (!is_separator(path[0]) && !(path[1] == ':' && is_separator(path[2]))) {
```

Comment:
This is correct, but a little subtle as it happens to also correctly handle UNC style paths which are absolute.

Reply:
I'll import ShLwApi and switch this to `!PathIsRelative()`.

Reply:
Sounds good